ec/ecresource/deploymentresource/schema_elasticsearch.go

```diff
@@ -79,24 +81,27 @@ func elasticsearchTopologySchema() *schema.Schema {
 	return &schema.Schema{
 		Type:        schema.TypeList,
 		MinItems:    1,
-		Required:    true,
-		Description: `Required topology element which must be set once but can be set multiple times to compose complex topologies`,
+		Optional:    true,
+		Computed:    true,
+		Description: `Optional topology element which must be set once but can be set multiple times to compose complex topologies`,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
 				"instance_configuration_id": {
 					Type:        schema.TypeString,
-					Description: `Required Instance Configuration ID from the deployment template`,
-					Required:    true,
+					Description: `Optional Instance Configuration ID from the deployment template`,
+					Computed:    true,
+					Optional:    true,
 				},
 				"memory_per_node": {
 					Type:        schema.TypeString,
 					Description: `Optional amount of memory per node in the "<size in GB>g" notation`,
-					Default:     "4g",
+					Default:     util.MemoryToState(defaultElasticsearchSize),
```

Comment:
optional nit: This could be more readable if it was set as a variable beforehand. Something like `defaultElasticsearchSizeGB := util.MemoryToState(defaultElasticsearchSize)`. What do you think?

Reply:
It wouldn't be my first choice, I left it as is to avoid too much indirection so we don't have an explosion of variables set in a place and then we have to go look what they are, if you feel very strongly about it I can change it

Reply:
nah, don't feel too strongly about it, we can leave it as is :)